dataclients/kubernetes/hosts.go

```diff
@@ -0,0 +1,70 @@
+package kubernetes
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/zalando/skipper/eskip"
+)
+
+func rxDots(h string) string {
+	return strings.Replace(h, ".", "[.]", -1)
+}
+
+func createHostRx(h ...string) string {
+	if len(h) == 0 {
+		return ""
+	}
+
+	hrx := make([]string, len(h))
+	for i := range h {
+		hrx[i] = rxDots(h[i])
+	}
+
+	return fmt.Sprintf("^(%s)$", strings.Join(hrx, "|"))
+}
+
+// hostCatchAllRoutes creates catch-all routes for those hosts that only have routes with
+// a Host predicate and at least one additional predicate.
+//
+// currently only used for RG
+func hostCatchAllRoutes(hostRoutes map[string][]*eskip.Route, createID func(string) string) []*eskip.Route {
+	var catchAll []*eskip.Route
+	for h, r := range hostRoutes {
+		var hasHostOnlyRoute bool
+		for _, ri := range r {
+			ct := eskip.Canonical(ri)
+			var hasNonHostPredicate bool
+			for _, p := range ct.Predicates {
+				if p.Name != "Host" {
```

Comment:
Maybe: `&& p.Name != "True"` ?

Reply:
not necessary, because then that would act as a catch-all, so in worst case we generated an unnecessary catch-all route. I'd rather wouldn't scatter the code with the True predicate, since I consider it rather a diagnostic tool, even if we use it currently for the traffic control. For the valid use cases, we need to find better solutions.

Reply:
Ok